### ISSUE TITLE
Implementation of the GraphSage layer

### ISSUE THREAD

**Issue by chrsmrrs** [pyg-team/pytorch_geometric#6]
Implementation of the GraphSage layer.

**Comment by zjhthu** [comment]
I think it should be `out = scatter_add(x[col], row, dim=0, dim_size=x.size(0))` rather than `out = scatter_add(x[row], row, dim=0, dim_size=x.size(0))` in the implementation of GraphSage layer?

**Comment by rusty1s** [comment]
Thank you! Fixed in master.

**Comment by luzai** [comment]
Thank you for your great repo! 

I think the graph_sage.py you implemented in `benchmark/kernel/` is `GraphSAGE-GCN` mentioned in the paper, without special design for mini-batch generation. I think mini-batch design for `GraphSAGE` is crucial for its scalability. 

May I ask whether there are some examples or benchmarks for training gcn on large-scale dataset? 

![image](https://user-images.githubusercontent.com/14788650/54826594-0d8ed200-4ceb-11e9-9a0d-a7b4e2aa25c4.png)

**Comment by rusty1s** [comment]
Yes, we currently implement GraphSAGE without neighborhood sampling. We have plans to integrate various sampling strategies (e.g. GraphSAGE, FastGCN, AdaptiveSampling) in the near future to enable training on large graphs.

**Comment by luzai** [comment]
Thank your very much for quick response! I am looking forward to the sampling strategies that enabling training on large graphs :)
